src/net.rs

```diff
@@ -183,34 +183,45 @@ impl Gossip {
     }
 
     /// Join a gossip topic with the default options and wait for at least one active connection.
-    pub async fn join(&self, topic_id: TopicId, bootstrap: Vec<NodeId>) -> Result<GossipTopic> {
-        let mut sub = self.join_with_opts(topic_id, JoinOptions::with_bootstrap(bootstrap));
+    pub async fn subscribe_and_join(
+        &self,
+        topic_id: TopicId,
+        bootstrap: Vec<NodeId>,
+    ) -> Result<GossipTopic> {
+        let mut sub = self.subscribe_with_opts(topic_id, JoinOptions::with_bootstrap(bootstrap));
         sub.joined().await?;
         Ok(sub)
     }
 
+    /// Join a gossip topic with the default options.
+    pub fn subscribe(&self, topic_id: TopicId, bootstrap: Vec<NodeId>) -> Result<GossipTopic> {
```

Comment:
note this will not wait for bootstrap nodes to be connected?

Reply:
yes, that's why there is now `subscribe_and_join`, to make this less confusing hopefully

Reply:
thanks